code/modules/atmospherics/portable/portable_atmospherics.dm

```diff
@@ -142,5 +142,33 @@
 			else
 				boutput(user, "<span class='notice'>Nothing happens.</span>")
 				return
-
+	else if (istype(W, /obj/item/reagent_containers/balloon))
+		var/obj/item/reagent_containers/balloon/balloon = W
+		var/amount = balloon.breaths * BREATH_VOLUME - TOTAL_MOLES(balloon.air)
+		if (amount <= 0)
+			boutput(user, "<span class='alert'>[balloon] is already full.</span>")
+			return
+		else if (istype(src, /obj/machinery/portable_atmospherics/scrubber))
+			var/fillfrom = tgui_alert(user, "Fill balloon from liquid tank or air tank?", "Fill balloon", list("Air tank", "Liquid tank", "Cancel"))
+			if (fillfrom == "Cancel")
+				return
+			else if (fillfrom == "Liquid tank")
+				if (!src.reagents.total_volume && src.reagents)
+					user.show_text("[src] is empty.", "red")
+					return
```

Comment:
```suggestion
				if (src.reagents && !src.reagents.total_volume)
					user.show_text("[src] is empty.", "red")
					return
```
This doesn't work as written, if `reagents` is null then the first statement will just cause an error since it's trying to access a var of `null`

Reply:
thanks for these!!! i'd copied this bit from balloons.dm and only swapped a few values around, so something like this would've never occurred to me